tests/snuba/api/endpoints/test_organization_events_spans_histogram.py

```diff
@@ -66,43 +61,255 @@ def create_event(self, **kwargs):
+        num_buckets = 50
+        query = {
+            "project": [self.project.id],
+            "span": self.format_span("django.middleware", "cd" * 8),
+            "numBuckets": num_buckets,
+        }
+
+        response = self.do_request(query)
+
+        assert response.status_code == 200, response.content
+        for bucket in response.data:
+            if bucket["bin"] == 3:
+                assert bucket["count"] == 2
+            elif bucket["bin"] == 10:
+                assert bucket["count"] == 3
+            else:
+                assert bucket["count"] == 0
+
+    def test_histogram_using_min_max(self):
         self.create_event()
-        num_buckets = 100
-
-        with self.feature(self.FEATURES):
-            response = self.client.get(
-                self.url,
-                data={
-                    "project": [self.project.id],
-                    "span": f"django.middleware:{'cd'* 8}",
-                    "numBuckets": num_buckets,
-                },
-                format="json",
-            )
-
-        # currently, num of buckets is hardcoded to 100, so we need to generate a result
-        # with 100 buckets
-        # then insert two non-null bins [{"bin": 3, "count": 2}, {"bin": 10, "count": 3}]
-        # TODO refactor this once num of buckets is dynamic
-        expected_data = [{"bin": i, "count": 0} for i in range(num_buckets)]
-        firstIndex = expected_data.index({"bin": 3, "count": 0})
-        expected_data[firstIndex] = {"bin": 3, "count": 2}
-        secondIndex = expected_data.index({"bin": 10, "count": 0})
-        expected_data[secondIndex] = {"bin": 10, "count": 3}
+        num_buckets = 10
+        min = 5
+        max = 11
+        query = {
+            "project": [self.project.id],
+            "span": self.format_span("django.middleware", "cd" * 8),
+            "numBuckets": num_buckets,
+            "min": min,
+            "max": max,
+        }
+
+        response = self.do_request(query)
 
         assert response.status_code == 200, response.content
-        assert response.data == expected_data
+        for bucket in response.data:
+            if bucket["bin"] == 10:
+                assert bucket["count"] == 3
+            else:
+                assert bucket["count"] == 0
+        assert response.data[0]["bin"] == min
+        assert response.data[-1]["bin"] == max - 1
+
+    def test_histogram_simple_using_given_min_above_queried_max(self):
+        self.create_event()
+        num_buckets = 10
+        min = 12
+        query = {
+            "project": [self.project.id],
+            "span": self.format_span("django.middleware", "cd" * 8),
+            "numBuckets": num_buckets,
+            "min": min,
+        }
+
+        response = self.do_request(query)
+
+        assert response.status_code == 200
+        for bucket in response.data:
+            assert bucket["count"] == 0
+        assert response.data[0] == {"bin": min, "count": 0}
```

Comment:
I'd also be curious to test what `max` is here, basically asserting what max became when the param is undefined

Reply:
interesting, so when `min` < `queried max` then only one bucket for that `min` is returned. this is consistent with how other histograms behave I believe, so i'm making an assertion here that there's only one element in the returned data